### ISSUE TITLE
Playground no longer works in Firefox when all cookies and storage is disabled

### ISSUE THREAD

**Issue by starblue** [integer32llc/rust-playground#431]
Playground no longer works for me, the window stays gray. I’m not exactly sure when and why this happened, maybe it is due to tightened security in some Firefox update in the last week or so. In the console I see the following security error:

```
Request to access cookie or storage on “https://play.integer32.com/” was blocked because we are blocking all storage access requests. 80926032aeb3aae9d0ff.js:1:27327
SecurityError: The operation is insecure. 80926032aeb3aae9d0ff.js:1
334          https://play.integer32.com/assets/80926032aeb3aae9d0ff.js:1:27328
i            https://play.integer32.com/assets/80926032aeb3aae9d0ff.js:1:514
176          https://play.integer32.com/assets/80926032aeb3aae9d0ff.js:1:7657
i            https://play.integer32.com/assets/80926032aeb3aae9d0ff.js:1:514
n            https://play.integer32.com/assets/80926032aeb3aae9d0ff.js:1:387
<anonymous>  https://play.integer32.com/assets/80926032aeb3aae9d0ff.js:1:3269
<anonymous>  https://play.integer32.com/assets/80926032aeb3aae9d0ff.js:1:2
```

Firefox version is 63.0.3 (64-bit) on Linux (Debian stable).
I use the uMatrix 1.3.14 and uBlock origin 1.17.2 add-ons, but disabling them doesn't change anything.

**Comment by shepmaster** [comment]
> I forbid that by default

How is this done? Without reporting how to *reproduce* an issue, there's really not much of a way that we can attempt to fix it.

**Comment by starblue** [comment]
This is in the end user visible part of the preferences:
Edit -> Preferences -> Privacy & Security -> Cookies and Site Data -> Block cookies and site data -> All cookies

**Comment by shepmaster** [comment]
> All cookies

Ah, the setting that specifically states that it will break websites?

![image](https://user-images.githubusercontent.com/174509/48981301-82d1f980-f0a1-11e8-968a-f9afac2e19c8.png)

**Comment by starblue** [comment]
Yes, and I can mostly deal with the consequences, i.e. explicitly enable cookies when they are needed, for example for logging in somewhere. I guess the warning is there for people who can't do that.
I don't think it a valid excuse for a website to break in an obscure way.

**Comment by inejge** [comment]
After #432, Playground doesn't work with cookies enabled (Firefox 63.0.3, 64-bit, Ubuntu 18.04).

```
TypeError: t is null 8cc65fe859fbd039fa13.js:1:28339
deserialize     https://play.rust-lang.org/assets/8cc65fe859fbd039fa13.js:1:28339
mn/</<          https://play.rust-lang.org/assets/8cc65fe859fbd039fa13.js:1:27661
d/</<           https://play.rust-lang.org/assets/2db6ccab5336953f0406.js:1:25975
u               https://play.rust-lang.org/assets/2db6ccab5336953f0406.js:1:21567
334             https://play.rust-lang.org/assets/8cc65fe859fbd039fa13.js:1:107309
i               https://play.rust-lang.org/assets/8cc65fe859fbd039fa13.js:1:514
176             https://play.rust-lang.org/assets/8cc65fe859fbd039fa13.js:1:7657
i               https://play.rust-lang.org/assets/8cc65fe859fbd039fa13.js:1:514
n               https://play.rust-lang.org/assets/8cc65fe859fbd039fa13.js:1:387
<anonymous>     https://play.rust-lang.org/assets/8cc65fe859fbd039fa13.js:1:3269
<anonymous>     https://play.rust-lang.org/assets/8cc65fe859fbd039fa13.js:1:2
```
After blocking the cookies (via an exception for https://play.rust-lang.org) it starts working.

**Comment by shepmaster** [comment]
> Playground doesn't work with cookies enabled

Should be fixed by #434; it's flowing through CI now.

**Comment by shepmaster** [comment]
> flowing through CI now.

The fix has been pushed; because the playground has stronger caching, you will either need to wait 24h or force a reload of the page.
